wp-dependency-installer.php

```diff
@@ -83,30 +110,37 @@ public function load_hooks() {
 		/**
 		 * Register wp-dependencies.json
 		 *
-		 * @param string $plugin_path Path to plugin or theme calling the framework.
+		 * @param string $plugin_path (optional) Path to plugin or theme calling the framework.
 		 */
-		public function run( $plugin_path ) {
+		public function run( $plugin_path = false ) {
+			if ( ! $plugin_path ) { // TODO: deprecate parameter in future releases?
+				$plugin_path = $this->plugin_path;
+			}
 			if ( file_exists( $plugin_path . '/wp-dependencies.json' ) ) {
 				$config = file_get_contents( $plugin_path . '/wp-dependencies.json' );
-				if ( empty( $config ) ||
-					null === ( $config = json_decode( $config, true ) )
-				) {
-					return;
-				}
-				$this->source = basename( $plugin_path );
-				$this->load_hooks();
-				$this->register( $config );
+				$config = json_decode( $config, true );
+			}
+			if ( ! empty( $config ) ) {
+				$this->register( $config, $plugin_path );
 			}
+			$this->load_hooks();
```

Comment:
Can you confirm that it is correct in this position? ( after the `$this->register( $config )` function ? )

Reply:
Since it's just loading hooks it just needs to go in `run()`, of course if there's no `$config` there's no point in running it.

Reply:
Probably should move into conditional checking that `! empty( $config )`